### PR TITLE
chore: Remove OpenCensus

### LINKED ISSUE #947: Remove code for Jaeger integration
### Describe request or inquiry 
The repo includes unused code and YAML configs related to Jaeger tracing, which has been replaced by showcasing Google Cloud Operations. As such, Jaeger-related code and documentation should be deprecated and removed.

### What purpose/environment will this feature serve? 
Since we now showcase Google CloudOps has our tracing agent, and we don't maintain a Jaeger variant of OnlineBoutique, it would make sense to remove the related code from the main stream.

From @askmeegs:
> The jaeger support in Online Boutique predates my time at Google, but my understanding is that it was designed as an OSS integration for tracing if the user didn't want to use Cloud Tracing (formerly Stackdriver Tracing). 
But right now the toggles for tracing enable/disable both Jaeger and Cloud Tracing at the same time. 🤷🏼 https://github.com/GoogleCloudPlatform/microservices-demo/blob/main/src/shippingservice/main.go#L206 
I think given that Online Boutique is a first party GCP demo designed to showcase GCP, it's okay to remove Jaeger from the upstream.

### PR DESCRIPTION
**Pull request by arbrown**
### Background 
In preparation for a proper OpenTelemetry integration, we have decided to remove existing references to OpenTelemetry to simplify development and maintenance toil related to maintaining deprecated dependencies.

### Fixes 
#947 
#635 

### Change Summary
Removes code and configuration related to OpenCensus and Jaeger

### Additional Notes
Only tracing and metrics are affected, and only in cases when users manually remove the DISABLE_TRACING and DISABLE_METRICS env variables from each service's manifest.  For now, there will still be no tracing and metrics, but soon these features will be implemented using OpenTelemetry.

### CODE REVIEW

File: docs/development-principles.md [hunk 1]
```diff
@@ -18,7 +18,7 @@ services. The experience of running the application locally or on GCP should
 be close.
 
 For example:
-- OpenCensus prints the traces to stdout when it cannot connect to GCP.
+- Tracing library prints the traces to stdout when it cannot connect to GCP.
```
Comment (bourgeoisor) [inline docs/development-principles.md:21]:
```suggestion
- The tracing library prints the traces to stdout when it cannot connect to GCP.
```

File: release/kubernetes-manifests.yaml [hunk 2]
```diff
@@ -144,8 +144,6 @@ spec:
             value: "1"
           - name: DISABLE_PROFILER
             value: "1"
-          # - name: JAEGER_SERVICE_ADDR
```
Comment (bourgeoisor) [inline release/kubernetes-manifests.yaml:147]: You don't want to manually commit changes to the `release` directory since that directory is linked to the latest release. In this case, you're making source code changes but those changes won't be part of the latest release (until a release happens). This file is automatically generated by https://github.com/GoogleCloudPlatform/microservices-demo/blob/main/hack/make-release.sh

TL;DR: revert the changes to this file.

File: src/adservice/src/main/java/hipstershop/AdService.java [hunk 3]
```diff
@@ -106,30 +93,19 @@ private static class AdServiceImpl extends hipstershop.AdServiceGrpc.AdServiceIm
...
-                  "Context Keys",
-                  AttributeValue.stringAttributeValue(req.getContextKeysList().toString()),
-                  "Context Keys length",
-                  AttributeValue.longAttributeValue(req.getContextKeysCount())));
           for (int i = 0; i < req.getContextKeysCount(); i++) {
             Collection<Ad> ads = service.getAdsByCategory(req.getContextKeys(i));
             allAds.addAll(ads);
           }
         } else {
-          span.addAnnotation("No Context provided. Constructing random Ads.");
```
Comment (bourgeoisor) [inline src/adservice/src/main/java/hipstershop/AdService.java:127]: Do you want to keep these sorts of messages (in a comment) so you remember to reimplement them with OpenTel?

File: src/checkoutservice/go.mod [hunk 4]
```diff
@@ -4,32 +4,22 @@ go 1.19
...
-	github.com/aws/aws-sdk-go v1.43.31 // indirect
-	github.com/census-instrumentation/opencensus-proto v0.3.0 // indirect
 	github.com/golang/groupcache v0.0.0-20210331224755-41bb18bfe9da // indirect
 	github.com/google/go-cmp v0.5.8 // indirect
 	github.com/google/pprof v0.0.0-20220412212628-83db2b799d1f // indirect
 	github.com/googleapis/gax-go/v2 v2.4.0 // indirect
-	github.com/jmespath/go-jmespath v0.4.0 // indirect
-	github.com/prometheus/prometheus v2.5.0+incompatible // indirect
-	github.com/uber/jaeger-client-go v2.25.0+incompatible // indirect
+	go.opencensus.io v0.23.0 // indirect
```
Comment (bourgeoisor) [inline src/checkoutservice/go.mod:21]: Where does this one come from?

File: src/frontend/go.mod [hunk 5]
```diff
@@ -5,34 +5,24 @@ go 1.19
...
-	github.com/census-instrumentation/opencensus-proto v0.3.0 // indirect
 	github.com/golang/groupcache v0.0.0-20210331224755-41bb18bfe9da // indirect
 	github.com/google/go-cmp v0.5.8 // indirect
 	github.com/google/pprof v0.0.0-20220412212628-83db2b799d1f // indirect
 	github.com/googleapis/enterprise-certificate-proxy v0.1.0 // indirect
 	github.com/googleapis/gax-go/v2 v2.4.0 // indirect
-	github.com/jmespath/go-jmespath v0.4.0 // indirect
-	github.com/prometheus/prometheus v2.5.0+incompatible // indirect
-	github.com/uber/jaeger-client-go v2.25.0+incompatible // indirect
+	go.opencensus.io v0.23.0 // indirect
```
Comment (bourgeoisor) [inline src/frontend/go.mod:24]: Where does this one come from?

File: src/productcatalogservice/go.mod [hunk 6]
```diff
@@ -17,18 +15,11 @@ require (
...
-	cloud.google.com/go/trace v1.0.0 // indirect
-	github.com/aws/aws-sdk-go v1.43.31 // indirect
-	github.com/census-instrumentation/opencensus-proto v0.3.0 // indirect
 	github.com/golang/groupcache v0.0.0-20210331224755-41bb18bfe9da // indirect
 	github.com/google/pprof v0.0.0-20220412212628-83db2b799d1f // indirect
 	github.com/googleapis/gax-go/v2 v2.4.0 // indirect
-	github.com/jmespath/go-jmespath v0.4.0 // indirect
-	github.com/prometheus/prometheus v2.5.0+incompatible // indirect
-	github.com/uber/jaeger-client-go v2.25.0+incompatible // indirect
+	go.opencensus.io v0.23.0 // indirect
```
Comment (bourgeoisor) [inline src/productcatalogservice/go.mod:21]: Ditto--

File: src/shippingservice/go.mod [hunk 7]
```diff
@@ -4,31 +4,21 @@ go 1.19
...
-	github.com/aws/aws-sdk-go v1.43.31 // indirect
-	github.com/census-instrumentation/opencensus-proto v0.3.0 // indirect
 	github.com/golang/groupcache v0.0.0-20210331224755-41bb18bfe9da // indirect
 	github.com/google/go-cmp v0.5.8 // indirect
 	github.com/google/pprof v0.0.0-20220412212628-83db2b799d1f // indirect
 	github.com/googleapis/gax-go/v2 v2.4.0 // indirect
-	github.com/jmespath/go-jmespath v0.4.0 // indirect
-	github.com/prometheus/prometheus v2.5.0+incompatible // indirect
-	github.com/uber/jaeger-client-go v2.25.0+incompatible // indirect
+	go.opencensus.io v0.23.0 // indirect
```
Comment (bourgeoisor) [inline src/shippingservice/go.mod:20]: Ditto--

File: src/adservice/src/main/java/hipstershop/AdService.java [hunk 8]
```diff
@@ -221,101 +197,29 @@ private static void initStats() {
       logger.info("Stats disabled.");
       return;
     }
-    logger.info("Stats enabled");
+    logger.info("Stats enabled, but temporarily unavailable");
```
Comment (bourgeoisor) [inline src/adservice/src/main/java/hipstershop/AdService.java:200]: Should we maybe link to the GitHub issue for these messages, e.g.:
"Stats enabled, but temporarily unavailable. See <URL> for details"

**Review by bourgeoisor — CHANGES_REQUESTED** [review]

File: src/frontend/go.mod (hunk 5 above)
Comment (arbrown) [reply src/frontend/go.mod:24]: According to ` go mod why -m go.opencensus.io` this comes from this chain:
```
github.com/GoogleCloudPlatform/microservices-demo/src/frontend
cloud.google.com/go/profiler
google.golang.org/api/transport/grpc
go.opencensus.io/plugin/ocgrpc
```
Though it's possible there are other dependency paths.  I imagine we'll address that in a future PR.

File: src/productcatalogservice/go.mod (hunk 6 above)
Comment (arbrown) [reply src/productcatalogservice/go.mod:21]: Same as above

File: src/checkoutservice/go.mod (hunk 4 above)
Comment (arbrown) [reply src/checkoutservice/go.mod:21]: Same as other comments.  Profiler Library.

File: src/adservice/src/main/java/hipstershop/AdService.java (hunk 3 above)
Comment (arbrown) [reply src/adservice/src/main/java/hipstershop/AdService.java:127]: I am planning on heavily referencing this PR during the implementation, so I will not miss details like this.

File: release/kubernetes-manifests.yaml (hunk 2 above)
Comment (arbrown) [reply release/kubernetes-manifests.yaml]: Done

File: docs/development-principles.md (hunk 1 above)
Comment (arbrown) [reply docs/development-principles.md]: Done

File: src/adservice/src/main/java/hipstershop/AdService.java (hunk 8 above)
Comment (arbrown) [reply src/adservice/src/main/java/hipstershop/AdService.java:200]: Done.

**Review by bourgeoisor — APPROVED** [review]
LGTM

File: README.md [hunk 9]
```diff
@@ -11,7 +11,7 @@ web-based e-commerce app where users can browse items,
 add them to the cart, and purchase them.
 
 **Google uses this application to demonstrate use of technologies like
-Kubernetes/GKE, Istio, Stackdriver, gRPC and OpenCensus**. This application
+Kubernetes/GKE, Istio, Stackdriver, and gRPC **. This application
```
Comment (NimJay) [inline README.md:14]:
```suggestion
Kubernetes/GKE, Istio, Stackdriver, and gRPC**. This application
```

File: README.md [hunk 10]
```diff
@@ -173,8 +173,6 @@ Find **Protocol Buffers Descriptions** at the [`./pb` directory](./pb).
 - **[gRPC](https://grpc.io):** Microservices use a high volume of gRPC calls to
   communicate to each other.
 - **[Istio](https://istio.io):** Application works on Istio service mesh.
-- **[OpenCensus](https://opencensus.io/) Tracing:** Most services are
```
Comment (NimJay) [inline README.md:176]: Praise: Thanks for paying attention to the littler details (such as documentation)!

File: src/recommendationservice/requirements.txt [hunk 11]
```diff
@@ -89,29 +81,14 @@ idna==3.4
...
     # via
     #   -r requirements.in
     #   google-api-core
     #   google-cloud-firestore
-    #   google-cloud-monitoring
     #   google-cloud-profiler
-    #   google-cloud-trace
     #   googleapis-common-protos
     #   grpcio-health-checking
     #   grpcio-status
```
Comment (NimJay) [inline src/recommendationservice/requirements.txt:94]: Praise: Thanks for re-generating the `requirements.txt` files!

**Review by NimJay — APPROVED** [review]
Amazing work, Drew! 
We're lucky to have been brought closer to the Cloud Observability DPEs.

I've gone through the changes. Looks good.
Thanks for also adding the note about "temporary" disablement (and including the link to the GitHub issue) — as Olivier's suggested.
And thank you, Olivier, for your review!

I've left one minor suggestion (in the `README.md` file).
Other than that, I approve.
Feel free to click "Squash and merge" whenever. 😄
